app/routes/messages.routes.js

```diff
@@ -0,0 +1,18 @@
+var messages = require('../controllers/messages.controller');
+var auth = require('../utils/auth-middleware.utils');
+
+module.exports = function (app) {
+    app.get('/messages/:listing_id/:first_user_id/:second_user_id',
+        auth.authenticateTokenMiddleware,
+        messages.findMessages
+    );
+
+    app.post('/messages',
+        auth.authenticateTokenMiddleware,
```

Comment:
This endpoint allows messages to be created between two users neither of whom is the owner of the listing. I'm not sure if you intended this?

Reply:
If not, then you can probably simplify this endpoint as well, through some assumptions. If you are not the Listing owner, then the message receiver will always be the owner of the listing.